addons/bus/models/bus_presence.py

```diff
@@ -31,6 +33,21 @@ class BusPresence(models.Model):
 
     @api.model
     def update(self, inactivity_period):
+        try:
+            self.with_context(MAX_TRIES_ON_CONCURRENCY_FAILURE=0)._update(inactivity_period)
+            # Commit on success.
+            # This method is called in _poll method and cursor is closed right after.
+            # See bus/controllers/main.py
+            self.env.cr.commit()
+            return True
+        except OperationalError as e:
+            if e.pgcode in PG_CONCURRENCY_ERRORS_TO_RETRY:
+                return False
```

Comment:
Here you catch the concurrency error, and return a normal value instead.
There will thus be no retry.
I suspect that putting `MAX_TRIES_ON_CONCURRENCY_FAILURE` in context is simply useless.

Reply:
Wait, shouldn't you rollback the transaction before returning?